src/aspire/basis/polar_2d.py

```diff
@@ -81,22 +82,19 @@ def evaluate(self, v):
         v, sz_roll = unroll_dim(v, 2)
         nimgs = v.shape[1]
 
-        half_size = self.nrad * self.ntheta // 2
+        half_size = self.ntheta // 2
 
         v = m_reshape(v, (self.nrad, self.ntheta, nimgs))
 
-        v = (v[:, :self.ntheta // 2, :]
-             + v[:, self.ntheta // 2:, :].conj())
+        v = (v[:, :half_size, :]
+             + v[:, half_size:, :].conj())
 
-        v = m_reshape(v, (half_size, nimgs))
+        v = m_reshape(v, (self.nrad*half_size, nimgs))
+        x = np.empty((self.sz[0], self.sz[1], nimgs), dtype='complex128', order='F')
```

Comment:
Why is this complex? Shouldn't it be real (with precision corresponding to that of `v`)? Also, is F-order needed here?

Reply:
Changed to float number and removed F-order.

Reply:
Yes but it needs to match the precision of `v`. If `v` is `complex128`, `x` should be `float64`. If `v` is `complex64`, `x` should be `float32`.

Reply:
Fixed it in commit  df7e893.